### PR TITLE
Only draw text on the side facing you!

### PR DESCRIPTION
**Pull request by Hackcraft**
This is a small edit which stops the text screen entity from drawing on both sides. With this, only the side facing you will be drawn.

`draw.DrawText` Was one of the most expensive calls on my client according to FProfiler, this halves the call count at the expense of a little maths.

### CODE REVIEW

File: lua/entities/sammyservers_textscreen/cl_init.lua [hunk 1]
```diff
@@ -60,32 +60,59 @@ function ENT:Draw()
...
+                      (entPos.y - plyShootPos.y) * direction.y +
+                      (entPos.z - plyShootPos.z) * direction.z
+    return (product < 0)
+end
+
+local plyShootPos, ang, pos, camangle, showFront -- Less variables being created each frame
 hook.Add( "PostDrawTranslucentRenderables", "SammyServers3D2DTextScreens", function()
+
+	-- Cache the shoot pos for this frame
+	if IsValid(LocalPlayer()) then
```
Comment (mcNuggets1) [inline lua/entities/sammyservers_textscreen/cl_init.lua]: localplayer can not be NULL at this time

Comment (Hackcraft) [reply lua/entities/sammyservers_textscreen/cl_init.lua]: Thank you.

**Comment by Cherry** [comment]
Interesting, thanks for the PR! Do you have any benchmarks or numbers showing the difference this brings in performance?

**Comment by Hackcraft** [comment]
In FProfiler, on a live DarkRP server with 4 permanent text screens with 4 lines each surrounding me and a couple from users. I profiled for 10(~0.02ms) seconds and got 202ms on the current version of text screens and 113ms on this version. However, on the test with this version, the drawing hook was called 18 times less (156 vs 174).

The average time for each call went down by 0.44ms (0.7259345092817ms vs 1.1633981738617ms).

**Comment by mcNuggets1** [comment]
3D2D drawing is ultra expensive for some reason.

**Comment by Cherry** [comment]
Thanks @Hackcraft! I'll review these changes and get them merged this week. 👍

**Comment by Cherry** [comment]
Merged and released, thanks!

**Pull request merged**